src/test/scala/org/coroutines/ast-canonicalization-tests.scala

```diff
@@ -90,7 +90,6 @@ class ASTCanonicalizationTest extends FunSuite with Matchers {
 
     val c = call(unit())
     assert(!c.resume)
-    assert(c.result == (()))
```

Comment:
Do you think it's fair to replace these calls with `assert(!c.hasException)`? If it is, we could avoid the compilation warnings.